src/SingleInstanceStateController.js

```diff
@@ -99,22 +100,16 @@ export function getObjectCache(obj: ObjectIdentifier): ObjectCache {
   return {};
 }
 
-export function estimateAttribute(obj: ObjectIdentifier, attr: string): mixed {
+export function estimateAttribute(obj: ParseObject, attr: string): mixed {
```

Comment:
Why was it necessary to change this to pass the object instead of the classname + id? This change causes many other refactors in this PR which should be avoided unless there is a benefit. If this change is unnecessary then I'd suggest to revert it and remove all the related refactors in this PRs.

Reply:
That's the whole point - just the object identifier (class name + id) is not sufficient in the unique instance mode to identify objects, which causes the sdk to lose track of dirty fields

Reply:
Got it, thanks